src/coalesce.c

```diff
@@ -0,0 +1,155 @@
+#include "data.table.h"
+
+SEXP coalesce(SEXP x, SEXP values) {
+  int JJ = LENGTH(values), nprotect = 0, NN = LENGTH(x);
+  
+  // pass over x in parallel to count NA
+  int na_count = 0;
+  switch(TYPEOF(x)) {
+  case LGLSXP: {
+    int *restrict x_ptr = LOGICAL(x);
+    #pragma omp parallel for schedule(dynamic) reduction(+:na_count)
```

Comment:
avoid `schedule` clause, auto should works good, and in this example I believe `static` would be better than `dynamic`

Reply:
also add `num_threads`

Reply:
we also need to measure overhead, small input and `system.time(for(i in 1:1000) coalesce(...))` as entering parallel might not be always useful, this can be easy to escape with extra omp clause `#pragma omp parallel for if (NN > 1000000) reduction(...) num_threads(...)`